server/repository/src/db_diesel/program_requisition/program_requisition_settings.rs

```diff
@@ -0,0 +1,202 @@
+use diesel::prelude::*;
+
+use crate::{
+    db_diesel::{
+        master_list_row::master_list::dsl as master_list_dsl,
+        name_tag_row::name_tag::dsl as name_tag_dsl,
+    },
+    program::dsl as program_dsl,
+    program_requisition_settings_row::program_requisition_settings::dsl as program_requisition_settings_dsl,
+    repository_error::RepositoryError,
+    MasterListFilter, MasterListRepository, MasterListRow, NameTagFilter, NameTagRepository,
+    ProgramRequisitionSettingsRow, ProgramRow, StorageConnection,
+};
+
+pub type ProgramRequisitionSettingsJoin =
+    (ProgramRequisitionSettingsRow, ProgramRow, MasterListRow);
+
+#[derive(Debug, PartialEq)]
+pub struct ProgramRequisitionSettings {
+    pub program_settings_row: ProgramRequisitionSettingsRow,
+    pub program_row: ProgramRow,
+    pub master_list: MasterListRow,
+}
+
+#[derive(Clone, PartialEq, Debug, Default)]
+pub struct ProgramRequisitionSettingsFilter {
+    pub name_tag: Option<NameTagFilter>,
+    pub master_list: Option<MasterListFilter>,
```

Comment:
Should we consider a Program Filter? I'm guessing master list is easier for some reason?

Reply:
Hmm, technically the filter is for program (for a program that's been joined to a store via master list which is joined to a store through master_list_name_join), I think master list filter is enough for now